src/main/java/objects/character/Boss.java

```diff
@@ -1,15 +1,15 @@
 package objects.character;
 
 import objects.battle.enemy.*;
-import objects.battle.enemy.gimmick.Gimmick;
+import objects.battle.enemy.gimmick.StatGimmickEntity;
 import objects.battle.enemy.ai.*;
 
 /**
  * This class represents the boss in a game
  */
 public class Boss extends Enemy {
 
-    private Gimmick gimmick;
+    private StatGimmickEntity gimmick;
```

Comment:
As discussed on discord, recommend to change storing an entity to storing a use case - that way you can implement some weird gimmicks like boss ignores type advantage without needing to replace the class or add extra logic - simply switch out the strategy for gimmicks.